### PR TITLE
Remove HistoryDepth to FinalityDepth validation

### PR DESCRIPTION
**Pull request by HelloKashif**
1. Remove HistoryDepth must be greater than FinalityDepth validation check
2. Update tests/docs

Context: https://chainlink-core.slack.com/archives/C03ANJB9VB7/p1714397285801799?thread_ts=1714395771.022789&cid=C03ANJB9VB7

### CODE REVIEW

File: core/gethwrappers/generated/automation_registry_logic_c_wrapper_2_3/automation_registry_logic_c_wrapper_2_3.go
Comment (dimriou) [inline core/gethwrappers/generated/automation_registry_logic_c_wrapper_2_3/automation_registry_logic_c_wrapper_2_3.go:1]: Is this file needed?

File: core/chains/evm/config/toml/config.go
Comment (dimriou) [inline core/chains/evm/config/toml/config.go:1]: Either use 3 different lines for Name, Value, and Msg to be consistent or revert back to the original formatting.

File: .changeset/healthy-plants-guess.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"chainlink": patch
+---
+
+#removed history depth to finality depth validation
```
Comment (dimriou) [inline .changeset/healthy-plants-guess.md]: This is not a NOP facing change, so maybe use #internal instead

File: core/services/chainlink/config_test.go [hunk 2]
```diff
@@ -1510,14 +1521,17 @@ func TestSecrets_Validate(t *testing.T) {
...
 		exp  string
 	}{
-		{name: "partial",
+		{
+			name: "partial",
 			toml: `
 Database.AllowSimplePasswords = true`,
 			exp: `invalid secrets: 2 errors:
 	- Database.URL: empty: must be provided and non-empty
-	- Password.Keystore: empty: must be provided and non-empty`},
```
Comment (dimriou) [inline core/services/chainlink/config_test.go]: Not sure if there is a specific style for these, but for such cases of string literals, I always found it easier to have the curly bracket along with the string because it avoids confusion as to whether a new line character exists at the end or not.

File: core/gethwrappers/generated/automation_registry_logic_c_wrapper_2_3/automation_registry_logic_c_wrapper_2_3.go
Comment (HelloKashif) [reply core/gethwrappers/generated/automation_registry_logic_c_wrapper_2_3/automation_registry_logic_c_wrapper_2_3.go:1]: I ran go generate for testing which generated it. Removed it from this PR to reduce confusion

File: core/chains/evm/config/toml/config.go
Comment (HelloKashif) [reply core/chains/evm/config/toml/config.go:1]: Removed all formatting changes to simplify the diff. Thanks

File: core/services/chainlink/config_test.go (hunk 2 above)
Comment (HelloKashif) [reply core/services/chainlink/config_test.go]: Yep, removed the formatting changes.

File: .changeset/healthy-plants-guess.md (hunk 1 above)
Comment (HelloKashif) [reply .changeset/healthy-plants-guess.md]: Done. Thanks

**Review by dimriou — APPROVED** [review]

**Review by dhaidashenko — APPROVED** [review]

**Review by simsonraj — APPROVED** [review]
